src/coreclr/nativeaot/docs/compiling.md

```diff
@@ -104,16 +104,45 @@ NativeAOT binaries built with this feature can run even when OpenSSL libraries a
 You can use this feature by adding the `StaticOpenSslLinking` property to your project file as follows:
 
 ```xml
-    <PropertyGroup>
-      <StaticOpenSslLinking>true</StaticOpenSslLinking>
-    </PropertyGroup>
+<PropertyGroup>
+  <StaticOpenSslLinking>true</StaticOpenSslLinking>
+</PropertyGroup>
 ```
 
 This feature is only supported on Linux. This feature is not supported when crosscompiling.
 
 ### Prerequisites
 
-Alpine
+Ubuntu
+```sh
+apt install libssl-dev cmake
 ```
+
+Alpine
+```sh
 apk add cmake openssl-dev openssl-libs-static
 ```
+
+## Using statically linked NUMA
+This feature can statically link NUMA library (libnuma.a) into your applications at build time.
+NativeAOT binaries built with this feature can run even when NUMA libraries are not installed.
+
+You can use this feature by adding the `StaticNumaLinking` property to your project file as follows:
```

Comment:
This should include warning about libnuma LGPL license. Static linking of LGPL code imposes additional requirements on the distribution of the final binary. See https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License.

Reply:
It is using v2.1 https://github.com/numactl/numactl/blob/master/LICENSE.LGPL2.1. IANAL, but it states:

>   When a program is linked with a library, whether statically or using
a shared library, the combination of the two is legally speaking a
combined work, a derivative of the original library.  The ordinary
General Public License therefore permits such linking only if the
entire combination fits its criteria of freedom.  The Lesser General
Public License permits more lax criteria for linking other code with
the library.
>
>  We call this license the "Lesser" General Public License because it
does Less to protect the user's freedom than the ordinary General
Public License.

Since, we are not shipping any of these bits statically linked; as an advisory, we can add a line linking to library license for each static library this document is discussing without interpretation of clauses.

Reply:
It is using v2.1 https://github.com/numactl/numactl/blob/master/LICENSE.LGPL2.1. IANAL, but it states:

>   When a program is linked with a library, whether statically or using
a shared library, the combination of the two is legally speaking a
combined work, a derivative of the original library.  The ordinary
General Public License therefore permits such linking only if the
entire combination fits its criteria of freedom.  The Lesser General
Public License permits more lax criteria for linking other code with
the library.
>
>  We call this license the "Lesser" General Public License because it
does Less to protect the user's freedom than the ordinary General
Public License.

Since, we are not shipping any of these bits statically linked; as an advisory, we can add a line linking to library license for each static library this document is discussing, without the interpretation of clauses.

Reply:
It is using v2.1 https://github.com/numactl/numactl/blob/master/LICENSE.LGPL2.1 (not v3, which is different). IANAL, but it states:

>   When a program is linked with a library, whether statically or using
a shared library, the combination of the two is legally speaking a
combined work, a derivative of the original library.  The ordinary
General Public License therefore permits such linking only if the
entire combination fits its criteria of freedom.  The Lesser General
Public License permits more lax criteria for linking other code with
the library.
>
>  We call this license the "Lesser" General Public License because it
does Less to protect the user's freedom than the ordinary General
Public License.

Since, we are not shipping any of these bits statically linked; as an advisory, we can add a line linking to library license for each static library this document is discussing as a heads up, without the interpretation of clauses.

Reply:
I agree we do not want to go into legal interpretation of LGPL. At the same time, we do want to highlight licensing traps that .NET users can fall into by following our documentation.

.NET project and many nuget packages have permissive license. Typically, one just needs to create THIRD-PARTY-NOTICES.TXT to full-fill licensing requirement for distributing binaries. The bar goes much higher once the binary starts statically linking LGPL code.

I have submitted suggested edit.